pyvista/plotting/theme.py

```diff
@@ -29,40 +29,99 @@ class Theme():
 
     def __init__(self):
         """Initialize the theme."""
-        self._jupyter_backend = None
-        self._auto_close = None
-        self._background = None
-        self._full_screen = None
-        self._camera = None
+        self._jupyter_backend = 'ipyvtklink'
+        self._auto_close = True  # DANGER: set to False with extreme caution
+        self._background = [0.3, 0.3, 0.3]
+        self._full_screen = False
+        self._camera = {
+            'position': [1, 1, 1],
+            'viewup': [0, 0, 1],
+        }
+
         self._notebook = None
-        self._window_size = None
-        self._font = None
-        self._cmap = None
-        self._color = None
-        self._nan_color = None
-        self._edge_color = None
-        self._outline_color = None
-        self._floor_color = None
-        self._colorbar_orientation = None
-        self._colorbar_horizontal = None
-        self._colorbar_vertical = None
-        self._show_scalar_bar = None
-        self._show_edges = None
-        self._lighting = None
-        self._interactive = None
-        self._render_points_as_spheres = None
-        self._use_ipyvtk = None
-        self._transparent_background = None
-        self._title = None
-        self._axes = None
-        self._multi_samples = None
+        self._window_size = [1024, 768]
+        self._font = {
```

Comment:
You changed themes from dict to a class with all its advantages (like doc-strings, parameter checking ...). What about the "sub-dicts" like font, camera, colorbars (two dicts horizontal/vertical with same structure), axes ... Maybe they could be replaced with objects too? Also, options with limited allowed values might be better represented by an Enum (like font-family, slider-style ...). Maybe also allow the "string"-solution as a fall-back for compatibility?

Reply:
I thought of this too.  It's becoming a bit of a painful PR to do everything right, but as I'm going through unit testing it's becoming apparent that some parameters really should be classed.